src/auth.ts

```diff
@@ -1,8 +1,48 @@
 import AinftBase from './ainftBase';
+import { INITIALIZE_STAKE_AMOUNT } from './constants';
 import { HttpMethod, User } from './types';
-
 export default class Auth extends AinftBase {
 
+  async initializeApp(appId: string, userId: string): Promise<void> {
+    const accessAccount = this.ain.wallet.defaultAccount!;
+    const createAppRes = await this.createApp(appId, userId, accessAccount.address);
+    console.log(`create app tx hash - ${createAppRes.txHash}`);
+    const stakeRes = await this.stake(appId, userId, INITIALIZE_STAKE_AMOUNT);
```

Comment:
To make things simple, can we make the staking is done for all the balance?

Reply:
It seems difficult to staking all the remaining balance because gas fee occurs in stake and setRule.
Is there a function to staking all remaining balance except gas fee in ain blockchain?

Reply:
I think the gas fee in the case is fixed. So what we can try is to try to stake (balance) - (expected gas fee)?

Reply:
That way, even if we change the initial ain amount, there will be no change in the here. I see!

Reply:
@woomurf exactly! 👍

Reply:
done!

Reply:
question) the rule set also consumes some gas fee?

Reply:
in my local server, it does not consume.  
It seems that I thought that the gas fee was charged by sending the transaction for stake and setRule at the same time.